scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java

```diff
@@ -1,53 +1,78 @@
 package com.avsystem.scex.util.function;
 
 import java.util.Collection;
+import com.avsystem.scex.presentation.annotation.Documentation;
 
 public interface StringUtil {
+    @Documentation("Returns concatenation of the provided comma-separated parts into a new string.")
     String concat(String... parts);
 
+    @Documentation("Returns true if the `list` of comma-separated values contains the string provided as the `item` argument.")
     boolean contains(String list, String item);
 
+    @Documentation("Returns true if the `source` string contains the value provided as `item`.")
     boolean stringContains(String source, String item);
 
+    @Documentation("Returns the first number in the provided string as a double value type. Returns 0 if string is null or does not include a number.")
     double extract(String string);
 
+    @Documentation("Returns a string which matches the regular expression `pattern`. Returns null if there is no match.")
     String regexFind(String value, String pattern);
 
+    @Documentation("Returns a group of characters which matches the regular expression `pattern`. Returns null if there is no match.")
     String regexFindGroup(String value, String pattern);
 
+    @Documentation("Returns true if the string `value` matches the regular expression `pattern`.")
     boolean regexMatches(String value, String pattern);
 
+    @Documentation("Returns a string with first part of the `value` string matching the regular expression `pattern` replaced with the `replacement` argument.")
     String regexReplace(String value, String pattern, String replacement);
 
+    @Documentation("Returns a string with all parts of the `value` string matching the regular expression `pattern` replaced with the `replacement` argument.")
     String regexReplaceAll(String value, String pattern, String replacement);
 
+    @Documentation("Returns all parts of a dot-separated string following the part indicated by `from` value, e.g. string.slice('a.b.c.d.', 2) returns 'c.d.'")
     String slice(String item, int from);
 
+    @Documentation("Returns all parts of a dot-separated string following the part indicated by `from` value. `Dot` boolean value determines if a dot is added to the output, e.g. string.slice('a.b.c.d.', 2, false) returns 'c.d'")
     String slice(String item, int from, int to, boolean dot);
 
+    @Documentation("Returns the `source` string with all non-alphanumeric characters replaced with the `replacement` string.")
     String stripToAlphanumeric(String source, String replacement);
 
+    @Documentation("Returns the `source` string with it last characters indicated by the `end` string removed.")
     String removeEnd(String source, String end);
 
+    @Documentation("Returns the `source` string with it first characters indicated by the `end` string removed.")
     String removeStart(String source, String start);
 
+    @Documentation("Returns the `source` string with `InternetGatewayDevice.` or `Device.` parts removed.")
```

Comment:
it might be worth mentioning that this method only removes the `InternetGatewayDevice.` or `Device.` strings from the beginning of the input